src/main/java/seedu/address/model/person/UniquePersonList.java

```diff
@@ -45,7 +47,14 @@ public void add(Person toAdd) {
         if (contains(toAdd)) {
             throw new DuplicatePersonException();
         }
-        internalList.add(toAdd);
+        int insertionIndex = 0;
+        for (Person person : internalList) {
+            if (toAdd.compareName(person) < 0) {
```

Comment:
`toAdd.getname().compare(person.getname())`

Reply:
I feel that this adheres to the abstraction principle better. The new code is covered under the existing test cases.